loki/expression/symbols.py

```diff
@@ -1457,3 +1460,80 @@ class StringSubscript(StrCompareMixin, pmbl.Subscript):
     @property
     def symbol(self):
         return self.aggregate
+
+class Reference(pmbl.Expression):
+    """
+    Internal representation of a Reference.
+
+    .. warning:: Experimental!
+
+    **C/C++ only**, no corresponding concept in Fortran.
+    Referencing refers to taking the address of an
+    existing variable (to set a pointer variable).
+    """
+    init_arg_names = ('expression',)
+
+    def __getinitargs__(self):
+        return (self.expression, )
+
+    def __init__(self, expression):
+        assert isinstance(expression, pmbl.Expression)
+        self.expression = expression
+
+    @property
+    def name(self):
+        return self.expression.name
+
+    @property
+    def type(self):
+        return self.expression.type
+
+    @property
+    def scope(self):
+        return self.expression.scope
+
+    @property
+    def initial(self):
+        return self.expression.initial
```

Comment:
I think it would be useful to leave a comment here that points out why this is necessary (e.g., something like that we allow the compound `[De]Reference(Variable(name))` to appear  with behaviour akin to a symbol itself for easier processing in mappers). Same in `Dereference`.